### PR TITLE
[SYCL] Use qualified name of template function call

### PR DESCRIPTION
**Pull request by smaslov-intel**
This is to fix https://github.com/intel/llvm/issues/1011.

The reason for name conflict is ADL (https://en.cppreference.com/w/cpp/language/adl)
The only reliable way to fix this is to use qualified name.

Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
Ping.

**Comment by smaslov-intel** [comment]
I will update the review with the test requested

File: sycl/include/CL/sycl/detail/pi.hpp [hunk 1]
```diff
@@ -134,8 +134,87 @@ template <typename Arg0, typename... Args>
...
+//    FnType  - Type of Function pointer to the PI API.
+//    FnOffset- Offset to the Function Pointer in the piPlugin::FunctionPointers
+//    structure. Used to differentiate between APIs with same pointer type,
+//    E.g.: piDeviceRelease and piDeviceRetain. Differentiation needed to avoid
+//    redefinition error during explicit specialization of class in pi.cpp.
+// Members: Initialized in default constructor in Class Template Specialization.
+// Usage:
+// Operator() - Call, Trace and Get result
+// Use Macro PI_CALL_NOCHECK call the constructor directly.
+template <typename FnType, size_t FnOffset> class CallPi {
```
Comment (AlexeySachkov) [inline sycl/include/CL/sycl/detail/pi.hpp]: I don't fully understand why these changes are present in the diff, is it a result of not fully correct merge from `sycl` branch?

File: sycl/test/regression/print_args.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+// Regression tests for https://github.com/intel/llvm/issues/1011
+// Checks that SYCL headers call internal templated function 'printArgs'
+// with fully quilified name to not confuse ADL.
+//
+//#include <iostream>
+template <typename TArg0, typename... TArgs>
+auto printArgs(TArg0 arg, TArgs... args)
+{
+    //std::cout << 42 << std::endl;
```
Comment (AlexeySachkov) [inline sycl/test/regression/print_args.cpp]: I suggest to either uncomment this line or to completely remove it

**Comment by bader** [comment]
@smaslov-intel, could you address review comments, please?

**Comment by smaslov-intel** [comment]
Sorry that it took ages to do this simple change. I think it is ready now.

**Comment by bader** [comment]
@smaslov-intel, the added regression test is empty. Could you upload the content of the test, please?

**Comment by garimagu** [comment]
The ADL seems to be looking up for functions in the local scope and in the scope of the arguments.
How is it looking into the global scope in the case highlighted by the test case?
We may see this issue with other functions too right?
